src/js/node/http2.ts

```diff
@@ -3354,10 +3354,32 @@ class ClientHttp2Session extends Http2Session {
         }
         headers[":scheme"] = scheme;
       }
+
       if (headers[":path"] == undefined) {
         headers[":path"] = "/";
       }
 
+      if (options !== undefined) {
+        if (!$isObject(options)) {
```

Comment:
```suggestion
        if (!$isObject(options) && !$isArray(options)) {
```

node uses assertIsObject which has a check for isArray https://github.com/nodejs/node/blob/1b5b12c3e6212d79884ae328c5710d4c9d2c2d8c/lib/internal/http2/util.js#L695. $isObject handles null, but will return true for arrays